src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioSymbolNavigationService.cs

```diff
@@ -119,14 +119,15 @@ public bool TryNavigateToSymbol(ISymbol symbol, Project project, OptionSet optio
             // Check whether decompilation is supported for the project. We currently only support this for C# projects.
             if (project.LanguageServices.GetService<IDecompiledSourceService>() != null)
             {
+                var eulaService = project.Solution.Workspace.Services.GetRequiredService<IDecompilerEulaService>();
```

Comment:
Is there any reason this a workspace service? Given the implementation is using JTF it means nobody can actually call it from any layer below the VS layer effectively, and I don't imagine different workspaces needing different implementations.

Reply:
The other IDE?

Reply:
Oh, well, technically, we would also be VS layer.

Reply:
@jasonmalinowski It uses `IVsPersistSolutionOpts`, which I assume is not available in other contexts

Reply:
Yeah, the method this is being called in is already using all sorts of other VS for Windows interfaces. :smile: